src/test/util_sds_check/util_sds_check.c

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2025, Hewlett Packard Enterprise Development LP */
+
+/*
+ * util_sds_check.c -- unit test for shutdown_state_check function
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include "unittest.h"
+#include "ut_pmem2.h"
+#include "shutdown_state.h"
+#include "set.h"
+
+FUNC_MOCK(util_is_zeroed, bool, void *ptr, size_t size)
+FUNC_MOCK_RUN_DEFAULT {
+	struct shutdown_state *sds = (struct shutdown_state *)ptr;
+	return (sds->uuid == 0 && sds->usc == 0 && \
+		sds->checksum == 0 && sds->dirty == 0);
+}
+FUNC_MOCK_END
+
+FUNC_MOCK(util_checksum, bool, struct shutdown_state *sds, size_t size, \
+	uint64_t *checksum, int a, int b)
+FUNC_MOCK_RUN_DEFAULT {
+	return sds->checksum == *checksum;
+}
+FUNC_MOCK_END
+
+FUNC_MOCK(shutdown_state_reinit, void, struct shutdown_state *curr_sds, \
+	struct shutdown_state *pool_sds, struct pool_replica *rep)
+FUNC_MOCK_RUN_DEFAULT {
+	*pool_sds = *curr_sds;
+}
+FUNC_MOCK_END
```

Comment:
1. The benefit of having this mock is that you can track whether it was called with `RCOUNTER(shutdown_state_reinit)`.
2. Other than that, this mock may just call the real implementation. An example how to do it below.

```suggestion
FUNC_MOCK(shutdown_state_reinit, void, struct shutdown_state *curr_sds, \
	struct shutdown_state *pool_sds, struct pool_replica *rep)
FUNC_MOCK_RUN_DEFAULT {
	_FUNC_REAL(shutdown_state_reinit)(curr_sds, pool_sds, rep);
}
FUNC_MOCK_END
```

Reply:
Your suggestion didn't work for me so I removed the mock.